clumper/clump.py

```diff
@@ -62,6 +62,42 @@ def read_json(cls, path):
             return Clumper(data)
         return Clumper(json.loads(pathlib.Path(path).read_text()))
 
+    @classmethod
+    def read_jsonl(cls, path, lines=None):
+        """
+        Reads in a jsonl file. You can also specify how many lines you want to read in.
+
+        """
+        data_array = []
+
+        if lines is not None:
+            assert lines >= 0, "Number of lines to read must be non-negative"
+
+        def iterate_over_file(file_handler):
+            nonlocal data_array, lines
+
+            for current_line_nr, json_string in enumerate(file_handler):
+                if lines is not None and current_line_nr == lines:
+                    break
+                json_object = json.loads(json_string)
+                data_array.append(json_object)
+            return data_array
+
+        try:
+            # Case 1 : Open cloud file in stream
+            if path.startswith("https:") or path.startswith("http:"):
+                with urllib.request.urlopen(path) as f:
+                    iterate_over_file(f)
+                    return Clumper(data_array)
+            # Case 2 : Local file
+            with open(path) as f:
+                iterate_over_file(f)
+                return Clumper(data_array)
```

Comment:
Potential simplification possible. I notice two blocks of; 

```python
iterate_over_file(f)
return Clumper(data_array)
```

Can't we use the if statment to decided how to grab the file reference `f` but from there apply the same operation?

Reply:
This way, you may not need the `iterate_over_file` anymore either.

Reply:
Yes, I can do this. Now that I am not using the requests library, I can simplify it.